packages/toolpad-app/src/components/AppEditor/AppEditorShell.tsx

```diff
@@ -109,6 +111,28 @@ export interface ToolpadAppShellProps {
 
 export default function AppEditorShell({ appId, ...props }: ToolpadAppShellProps) {
   const domLoader = useDomLoader();
+  const dom = useDom();
```

Comment:
I don't think that reasoning is correct. It's not "first visit to the Editor", it's "visit the root of the editor while tehre exists pages". I'm not sure this logic belongs in this place. I think you'll need to use a [`Navigate`](https://reactrouter.com/docs/en/v6/components/navigate) component under the `/` route, here https://github.com/mui/mui-toolpad/blob/0f703b1252920837c3f8843053e19e6840d27f58/packages/toolpad-app/src/components/Studio.tsx#L18

Something like

```tsx
<Route path="/" element={hasPages ? <Navigate to={firstPageUrl} /> : <Home />} />
```

Reply:
Using the `useDom` and `useLocation` hooks inside the `<Studio>` component may not be feasible, since the dom isn't completely loaded by the time the component is rendered, and `useLocation` is only allowed within the context of a `<Router>` component